feature/experimental/bgp/ate_tests/bgp_long_lived_graceful_restart/bgp_long_lived_graceful_restart_test.go

```diff
@@ -0,0 +1,1333 @@
+		fptest.AssignToNetworkInstance(t, dut, i1.GetName(), deviations.DefaultNetworkInstance(dut), 0)
+		fptest.AssignToNetworkInstance(t, dut, i2.GetName(), deviations.DefaultNetworkInstance(dut), 0)
+	}
+}
+
+func verifyPortsUp(t *testing.T, dev *ondatra.Device) {
+	t.Helper()
+	for _, p := range dev.Ports() {
+		status := gnmi.Get(t, dev, gnmi.OC().Interface(p.Name()).OperStatus().State())
+		if want := oc.Interface_OperStatus_UP; status != want {
+			t.Errorf("%s Status: got %v, want %v", p, status, want)
+		}
+	}
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+}
+
+func buildNbrList(asN uint32) []*bgpNeighbor {
+	nbr1v4 := &bgpNeighbor{as: asN, neighborip: atePort1SubIntf1.IPv4, isV4: true}
+	nbr1v6 := &bgpNeighbor{as: asN, neighborip: atePort1SubIntf1.IPv6, isV4: false}
+	nbr2v4 := &bgpNeighbor{as: asN, neighborip: ateDst.IPv4, isV4: true}
+	nbr2v6 := &bgpNeighbor{as: asN, neighborip: ateDst.IPv6, isV4: false}
+	return []*bgpNeighbor{nbr1v4, nbr2v4, nbr1v6, nbr2v6}
+}
+
+func bgpWithNbr(as uint32, nbrs []*bgpNeighbor, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni_proto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := ni_proto.GetOrCreateBgp()
+
+	g := bgp.GetOrCreateGlobal()
+	g.As = ygot.Uint32(as)
+	g.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Enabled = ygot.Bool(true)
+	g.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Enabled = ygot.Bool(true)
+	g.RouterId = ygot.String(dutDst.IPv4)
+	bgpgr := g.GetOrCreateGracefulRestart()
+	bgpgr.Enabled = ygot.Bool(true)
+	bgpgr.RestartTime = ygot.Uint16(grRestartTime)
+	bgpgr.StaleRoutesTime = ygot.Uint16(grStaleRouteTime)
+
+	pg := bgp.GetOrCreatePeerGroup(peerv4GrpName)
+	pg.PeerAs = ygot.Uint32(ateAS)
+	pg.PeerGroupName = ygot.String(peerv4GrpName)
+
+	pgv6 := bgp.GetOrCreatePeerGroup(peerv6GrpName)
+	pgv6.PeerAs = ygot.Uint32(ateAS)
+	pgv6.PeerGroupName = ygot.String(peerv6GrpName)
+
+	if deviations.RoutePolicyUnderAFIUnsupported(dut) {
+		rpl := pg.GetOrCreateApplyPolicy()
+		rpl.SetExportPolicy([]string{"ALLOW"})
+		rpl.SetImportPolicy([]string{"ALLOW"})
+		rplv6 := pgv6.GetOrCreateApplyPolicy()
+		rplv6.SetExportPolicy([]string{"ALLOW"})
+		rplv6.SetImportPolicy([]string{"ALLOW"})
+	} else {
+		pg1af4 := pg.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+		pg1af4.Enabled = ygot.Bool(true)
+		pg1rpl4 := pg1af4.GetOrCreateApplyPolicy()
+		pg1rpl4.SetExportPolicy([]string{"ALLOW"})
+		pg1rpl4.SetImportPolicy([]string{"ALLOW"})
+		pg1af6 := pgv6.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST)
+		pg1af6.Enabled = ygot.Bool(true)
+		pg1rpl6 := pg1af6.GetOrCreateApplyPolicy()
+		pg1rpl6.SetExportPolicy([]string{"ALLOW"})
+		pg1rpl6.SetImportPolicy([]string{"ALLOW"})
+	}
+
+	for _, nbr := range nbrs {
+		if nbr.isV4 {
+			nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
```

Comment:
the if- and else- branches have a lot in common. Why don't you create the neighbor and set the common parameters before the if-statement, i.e.

```
neighbor := bgp.GetOrCreateNeighbor(nbr.neighborip)
neighbor.GetOrCreateTimers().HoldTime = ygot.Uint16(180)
neighbor.GetOrCreateTimers().KeepaliveInterval = ygot.Uint16(60)
neighbor.PeerAs = ygot.Uint32(nbr.as)
neighbor.Enabled = ygot.Bool(true)

if nbr.isV4 {
...
} else {
...
}

```

Reply:
updated, thank you